gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcReadChannel.java

```diff
@@ -379,9 +377,15 @@ private static ByteString getFooterContent(
     }
   }
 
+  // Estimates a read time out based on read speeds
+  private static long getReadTimeoutMillis(
+      GoogleCloudStorageReadOptions readOptions, long objectSize) {
```

Comment:
`readOptions` is a object variable. do we need to pass it along in multiple functions calls?

Reply:
This method is also invoked in the `Open` api path, where the object is not initialised. and we need the method to be statid.